### PR TITLE
Update typescript recipe

### LINKED ISSUE #1822: Update TypeScript recipe given new extensions support
It'd be good to update the TypeScript recipe in light of AVA's new `extensions` support.

I'm suggesting the following in the release notes, but I must admit I haven't actually tested it, and it's not mentioned in the recipe yet:

```json
{
    "ava": {
        "compileEnhancements": false,
        "extensions": ["ts"],
        "require": [
            "ts-node/register"
        ]
    }
}
```

Shout if you're up for trying the snippet and updating the recipe 😄

### PR DESCRIPTION
**Pull request by btkostner**
Fixes #1822

I removed the translation links, because it's such a large update. Should I undo this?

TODO:
- [x] Get code coverage working

### CODE REVIEW

File: docs/recipes/typescript.md [hunk 1]
```diff
@@ -1,29 +1,47 @@
...
+incorrect type errors, or other _non critical_ errors, you can use
+`ts-node/register/transpile-only` instead.
 
 ## Writing tests
 
 Create a `test.ts` file.
 
 ```ts
-import test from 'ava';
+import { test } from 'ava';
```
Comment (novemberborn) [inline docs/recipes/typescript.md]: Does `import test from 'ava'` not work?

File: docs/recipes/typescript.md [hunk 2]
```diff
@@ -1,29 +1,47 @@
...
+
+And last but not least, configure ava to build typescript files by adding
+this configuration to your `package.json` file.
 
 ```json
-{
-	"scripts": {
-		"test": "tsc && ava"
+	"ava": {
+		"babel": false,
```
Comment (novemberborn) [inline docs/recipes/typescript.md]: Hopefully we can land https://github.com/avajs/ava/issues/1836 first, and then this should be removed.

**Comment by novemberborn** [comment]
Let's keep the translation links. Hopefully people notice that they're looking at outdated documentation.

Formatting wise, we don't put in hard wraps in the Markdown files for this project. There seem to be a lot of unnecessary changes due to this. Could you revert those?

File: docs/recipes/typescript.md (hunk 1 above)
Comment (btkostner) [reply docs/recipes/typescript.md]: It does, but I felt that `import { test } from 'ava';` was more consistent considering that's how it's used later in the docs. Also, a lot of the typescript community (opinionated and no numbers to back this up) recommend against using `export default`.

**Comment by btkostner** [comment]
My bad. A lot of those edits where me being OCD. I'll revert those tonight when I get a chance.

File: docs/recipes/typescript.md (hunk 1 above)
Comment (novemberborn) [reply docs/recipes/typescript.md]: We show the default export everywhere else so let’s keep doing the same here.

**Comment by btkostner** [comment]
Alright. It looks good to me. One thing to note, I was unable to get `nyc` code coverage to work correctly with this setup, but that's not a show stopper for this PR.

**Comment by btkostner** [comment]
Never mind on my last message. Apparently it's working on travis, so something locally is funky with my setup.

**Comment by novemberborn** [comment]
> I was unable to get nyc code coverage to work correctly with this setup, but that's not a show stopper for this PR.
>
> Never mind on my last message. Apparently it's working on travis, so something locally is funky with my setup.

Yea that stuff gets real difficult real quick. Thanks for looking into it though!

**Pull request merged**